File: JavaSolutions/src/_0009Solution.java
// https://leetcode.com/problems/palindrome-number/
// medium
// Time: O(log x); Space: O(1) 

public class _0009Solution {

  public static void main(String[] args) {
    int i1 = 121;
    int i2 = 122;
    int i3 = -111;

    System.out.println(isPalindrome(i1));
    System.out.println(isPalindrome(i2));
    System.out.println(isPalindrome(i3));
  }

  private static boolean isPalindrome(int x) {
    if(x < 0) return false;

    int y = x;
    int flip = 0;
    while(y != 0) {
      flip = flip * 10 + y % 10;
      y /= 10;
    }

    return flip == x;
  }

}

File: JavaScriptSolutions/_0009Solution.js
// https://leetcode.com/problems/palindrome-number/
// Easy
// 234, 2217, 2396

/**
 * @param {number} x
 * @return {boolean}
 */
var isPalindrome = function (x) {
  const isNegative = x < 0 ? true : false;
  if (isNegative) return false;
  let temp = x;
  let reverse = 0;
  while (temp > 0) {
      reverse = (reverse * 10) + (temp % 10)
      temp = parseInt(temp / 10)
  }

  return reverse == x;
};